MapboxNavigation/NavigationViewController.swift

```diff
@@ -17,7 +17,7 @@ public protocol NavigationViewControllerDelegate {
     /**
      Called when the user arrives at the destination.
      */
-    @objc optional func navigationViewController(_ navigationViewController : NavigationViewController, didArriveAt waypoint: Waypoint)
+    @objc optional func navigationViewController(_ navigationViewController : NavigationViewController, didArriveAt waypoint: Waypoint, finalDestination: Bool)
```

Comment:
Would it be possible to leave this unchanged and check `waypoint == route.routeOptions.waypoints.last`?

Reply:
What if we keep this function the same but just check `routeProgress.remainingWaypoints` instead?

Reply:
- [ ] Check for last waypoint using `navigationViewController` argument instead of exposing it as an argument